### PR TITLE
Change comparison in partitions to include equals

### PR DESCRIPTION
**Pull request by JAEarly**
## Description of changes:

A warning is raised during partitioning when the number of canonical nodes is equal to the number of samples. It is my understanding that this should be valid (one sample per node), and that issues only occur when the number of samples is less than the number of canonical nodes. Apologies if my understanding is incorrect here. 

The simple fix is to the change the comparison to be `>=` rather than `>`

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [ ] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [x] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [x] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#submitting-a-contribution))
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

**Comment by JAEarly** [comment]
I didn't realise this would break the tests (I made the change in the browser). If this change makes sense I'll happily go and alter the tests as appropriate.

**Comment by JAEarly** [comment]
I've now updated the code such that the tests pass but the warning is not raised for the edge case: num_canonical_nodes == num_samples and num_canonical_nodes > num_physical_nodes.

File: streaming/base/partition/orig.py [hunk 1]
```diff
@@ -81,7 +81,8 @@ def get_partitions_orig(num_samples: int,
             padding = node_ratio - overflow
     padded_samples_per_canonical_node = samples_per_canonical_node + padding
 
-    if num_samples > num_canonical_nodes:
+    if num_samples > num_canonical_nodes or (num_samples == num_canonical_nodes and
+                                             num_canonical_nodes > num_physical_nodes):
```
Comment (snarayan21) [inline streaming/base/partition/orig.py:85]: I believe the condition here can actually be slightly expanded to `num_canonical nodes >= num_physical nodes`. So the full conditional would be:
```
if num_samples > num_canonical_nodes or (num_samples == num_canonical_nodes and num_canonical_nodes >= num_physical_nodes):
```
Mind making this change and adding an explanatory comment right before the if statement?

**Review by snarayan21 — COMMENTED** [review]
Hey! Apologies for getting to this late, but just a small change and should be good to go. Great catch!

File: streaming/base/partition/orig.py (hunk 1 above)
Comment (JAEarly) [reply streaming/base/partition/orig.py]: Thank you for the input; I've made that change now.

**Review by snarayan21 — APPROVED** [review]
lgtm!
